### DISCUSSION TITLE
Can't install on FreeBSD

### DISCUSSION THREAD

**Discussion by scorpio1441** [openai/whisper#901] [General]
Fresh instance on AWS as simple as that:

# pkg install py39-pip
............
# pip install -U openai-whisper
Collecting openai-whisper
  Downloading openai-whisper-20230124.tar.gz (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 39.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting numpy
  Downloading numpy-1.24.1.tar.gz (10.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.9/10.9 MB 45.6 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting openai-whisper
  Downloading openai-whisper-20230117.tar.gz (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 54.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
**ERROR: Cannot install openai-whisper==20230117 and openai-whisper==20230124 because these package versions have conflicting dependencies.**

The conflict is caused by:
    openai-whisper 20230124 depends on torch
    openai-whisper 20230117 depends on torch

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

**ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts**

**Comment by glangford** [comment]
> ERROR: Cannot install openai-whisper==20230117 and openai-whisper==20230124 

Not sure why it is trying to install two versions of `whisper`. Have you tried
`pip uninstall openai-whisper`
`pip install -vv "openai-whisper==20230124"`

Please share the verbose log from the install if that doesn't work.

**Reply by scorpio1441** [reply]
Same error, please see:
[verbose.txt](https://github.com/openai/whisper/files/10611211/verbose.txt)

**Reply by glangford** [reply]
The problem is that PyTorch cannot be installed. It is a requirement for `whisper`.

Near the bottom of the log,
> ERROR: Could not find a version that satisfies the requirement torch (from openai-whisper) (from versions: none)
> ERROR: No matching distribution found for torch

According to this post, PyTorch is not supported on FreeBSD
https://discuss.pytorch.org/t/how-to-pytorch-install-on-freebsd-12-1/169582

**Reply by scorpio1441** [reply]
Is PyTorch absolutely required?
It’s available as C++ package, maybe it’s possible to use that instead.

**Reply by glangford** [reply]
There is a fork of `whisper` using C/C++, see here
* https://github.com/ggerganov/whisper.cpp#readme
* https://github.com/ggerganov/whisper.cpp/issues/56#issuecomment-1350920264

**Reply by scorpio1441** [reply]
whisper.cpp is what I'm switching from =)))

**Reply by glangford** [reply]
Ok then! :)
To answer your question then, this whisper absolutely requires PyTorch.

**Reply by nkfilis** [reply]
there now is a FreeBSD port: https://www.freshports.org/misc/py-pytorch/
